tests/test_submit.py

```diff
@@ -0,0 +1,122 @@
+                internet_permissions=[],
+                git_push_result=None,
+                expected_prompts_start=[
+                "Since this task is running on a container with no internet access,"
+                ],
+                expected_output=[
+                    "No uncommitted changes in"
+                ],
+            ),
+            id="no_changes_without_internet",
+        ),        
+        pytest.param(
+            CheckoutGitTestScenario(
+                git_status_mocked_output="M modified_file.txt",
+                internet_permissions=[],
+                git_push_result=None,
+                expected_prompts_start=[
+                "Are you sure you want to continue?",
+                "Since this task is running on a container with no internet access,",
+                ],
+                expected_output=[
+                    "Uncommitted changes in",
+                    "M modified_file.txt"
+                ],
+            ),
+            id="uncommitted_changes_confirmed",
+        ),        
+        pytest.param(
+            CheckoutGitTestScenario(
+                internet_permissions=["internet"],
+                git_push_result=(1, "Failed to push: remote error"),
+                expected_prompts_start=["Are you sure you want to continue?"],
+                expected_output=[
+                    "No uncommitted changes in",
+                    "Failed to push to git remote:",
+                    "Failed to push: remote error"
+                ],
+            ),
+            id="push_failure_confirmed",
+        ),
+    ]
+)
+async def test_check_git_repo(
+    mocker: MockerFixture,
+    capsys: pytest.CaptureFixture,
+    fp: FakeProcess,
+    scenario: CheckoutGitTestScenario,
+):
+    # Mock settings.get_settings (before importing submit)
+    mock_settings = mocker.Mock(permissions=scenario.internet_permissions)
+    mocker.patch("src.settings.get_settings", return_value=mock_settings, autospec=True)
+
+    from src.submit import _check_git_repo
+
+    # Mock git status using pytest-subprocess
+    fp.register(
+        ["git", "status", "--porcelain"],
+        stdout=scenario.git_status_mocked_output.encode()
+    )
+
+    # Mock git push if needed
+    if scenario.git_push_result is not None:
+        exit_code, output = scenario.git_push_result
+        fp.register(
+            ["git", "push"],
+            stdout=output.encode(),
+            returncode=exit_code
+        )
+
+    # Mock click.confirm and track calls
+    mock_confirm = mocker.patch("click.confirm", return_value=True, autospec=True)
+
+    # Run the function
+    repo_dir = pathlib.Path("/fake/path")
+    await _check_git_repo(repo_dir)
+
+    # Verify the confirmation prompts
+    actual_prompts = [call.args[0] for call in mock_confirm.call_args_list]
+    assert all(actual_prompt.startswith(expected_prompt) for actual_prompt, expected_prompt in zip(actual_prompts, scenario.expected_prompts_start))
```

Comment:
```suggestion
    assert mock_confirm.call_count == len(scenario.expected_prompts_start)
    assert all(
        expected_prompt in call.args[0]
        for call, expected_prompt in zip(
            mock_confirm.call_args_list, scenario.expected_prompts_start
        )
    )
```

Reply:
I like the `len` test. I don't like that this won't use the word `actual`. I like seeing expected vs actual in test results